OpenSim/Common/Component.cpp

```diff
@@ -1524,6 +1504,31 @@ void Component::dumpConnections() const {
     }
 }
 
+void Component::printSubcomponentInfo() const {
+    printSubcomponentInfo<Component>();
+}
+
+void Component::printOutputInfo(const bool includeDescendants) const {
+    // Do not display header for Components with no outputs.
+    if (getNumOutputs() > 0) {
+        const std::string msg = "Outputs from " + getAbsolutePathName();
+        std::cout << msg << "\n" << std::string(msg.size(), '=') << std::endl;
+
+        std::vector<std::string> outputNames = getOutputNames();
+        for (auto thisName : outputNames)
+            std::cout << "  " << thisName << std::endl;
+        std::cout << std::endl;
+    }
+
+    if (includeDescendants) {
+        for (const Component& thisComp : getComponentList<Component>()) {
+            // compList (comp's ComponentList) includes all descendants (i.e.,
```

Comment:
There's no variable named `compList`. I'm guessing that you modified the code after C&P but didn't make the corresponding change to the comment.